osxfuse/fuse_fsevents.c

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2016 EditShare LLC
+ * All rights reserved.
+ */
+
+#include "fuse_fsevents.h"
+
+#if VERSION_MAJOR >= 15
+
+void
+fsevent(vnode_t vp, uint32_t hint)
+{
+    struct vnode_attr vattr;
+
+    vfs_get_notify_attributes(&vattr);
+    vnode_notify(vp, hint, &vattr);
```

Comment:
You are actually supposed to fill in the active attributes before calling `vnode_notify()`. To make things worse vattr consists of mostly uninitialized memory. This can result in undefined behavior and kernel panics.